### PR TITLE
Feature/postpone update on network disconnection

### PR DESCRIPTION
**Pull request by jaelim-ms**
Want to share the implementation earlier. Unit test is coming soon.

### CODE REVIEW

File: MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m [hunk 1]
```diff
@@ -48,13 +48,13 @@ - (instancetype)init {
   if ((self = [super init])) {
     _apiUrl = kMSDefaultApiUrl;
     _installUrl = kMSDefaultInstallUrl;
-    NSNumber *flag = [MS_USER_DEFAULTS objectForKey:kMSSDKHasLaunchedWithDistribute];
 
     /*
      * Delete API token if an application has been uninstalled and try to get a new one from server.
      * Under iOS 10.3, keychain data won't be automatically deleted by uninstall
```
Comment (TroubleMakerBen) [inline MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m]: This comment is misleading "under 10.3" reads to me as if this was the case on 10.3.
Maybe say "for OS version < 10.3" or something.

File: MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m [hunk 2]
```diff
@@ -134,6 +134,14 @@ - (void)requestUpdateToken {
   // Check if it's okay to check for updates.
   if ([self checkForUpdatesAllowed]) {
 
+    // Check if the device has internet connection to get update token.
+    if ([MS_Reachability reachabilityForInternetConnection].currentReachabilityStatus == NotReachable) {
+      MSLogWarning(
+          [MSDistribute logTag],
+          @"The device lost internet connection. The SDK will retry to get update API token in the next launch.");
```
Comment (TroubleMakerBen) [inline MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m]: missing `it's` before "internet connection" and "an" before "update-API token".

Comment (jaelim-ms) [inline MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m]: Done. And I think `it's` -> `its`.

File: MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m (hunk 1 above)
Comment (jaelim-ms) [inline MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m]: Done.

**Pull request merged**